### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by maths29**
In this project, i created a form of UI for the school library app. This way it can be invoked as an executable and not something you use in IRB exclusively.
- Build interactive console apps.
- Run a program using the command line.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require_relative 'app'
+
+def main
+  app = App.new
+  app.start
+end
+
+main
```
Comment (algerina) [inline main.rb]: - [ ] While you made a big effort working on the entire App, you still have an issue to work on. Your App didn't work correctly I have got the following issue and I recommend checking the files and class names:

Screenshot 2023-09-29 at 5 50 33 PM

**Review by algerina — CHANGES_REQUESTED** [review]
Hi @maths29 ,

Great job so far! However you have some minor issues to work on.

May I highlight the following: 
-  Your components are almost setup correctly✔
- The code is clean and well organized✔️



## CHANGES REQUIRED:muscle::clap:


Check the comments under the review.



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    if user_type == 1
+      print 'Has parent permission? [Y/N]: '
+      permission = gets.chomp.strip.upcase == 'Y'
+      @people << Student.new(age, @classroom, permission, name)
+    else
+      print 'Specialization: '
+      specialization = gets.chomp.strip
+      @people << Teacher.new(age, specialization, name)
+    end
+  end
```
Comment (levy002) [inline app.rb]: - [ ] Kindly ensure the user can create the `Person` successfully. When creating a new person there is an error that blocks a user from creating a new person successfully. Kindly fix the issue.

### ERROR

   image

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @maths29    , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter error :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  # Method to create a person (teacher or student)
+  def create_student(name, age, parent_permission)
+    student = Students.new(name, age, parent_permission)
+    @persons << student
+  end
+
+  def create_teacher(name, age, specialization)
+    teacher = Teacher.new(name, age, specialization)
+    @persons << teacher
+  end
```
Comment (brainconnect93) [inline app.rb:31]: - [ ] Great effort working on the suggestion requested by the previous reviewer, but currently, you only work on creating one of the person, which is the student, but when I tried to create a teacher I was getting an error. I would kindly suggest you work on it and be sure to have the app working perfectly

Screenshot 👇 

oop

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @maths29, 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- Rubocop Linters setup properly✔️
- School Library App properly setup ✔️
- App works really great ✔️
- Add the basic UI for the School library app ✔️
- Good use of Git flow ✔️ 

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @maths29 👏 👏 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Console app worked properly as expected ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**